apps/src/blockly/customBlocks/googleBlockly/mutators/functionMutatorHelpers.js

```diff
@@ -0,0 +1,38 @@
+// Common helper functions for procedure and behavior mutators.
+import msg from '@cdo/locale';
+
+// In Lab2, the level properties are in Redux, not appOptions. To make this work in Lab2,
+// we would need to send that property from the backend and save it in lab2Redux.
+const useModalFunctionEditor = window.appOptions?.level?.useModalFunctionEditor;
+
+// If the definition block has a description property
+// in extra state, either add a new editable input field to the block
+// for the description (if the modal function editor is enabled),
+// or set the description property on the block (if the modal function editor is disabled).
+export function handleLoadDescription(block, extraState) {
+  const description = extraState['description'];
+  if (useModalFunctionEditor) {
+    block
+      .appendDummyInput('DESCRIPTION_ROW')
+      .appendField(`${msg.description()} `, 'DESCRIPTION_LABEL')
+      .appendField(new Blockly.FieldTextInput(description), 'DESCRIPTION');
+    block.moveInputBefore('DESCRIPTION_ROW', 'STACK');
```

Comment:
I was going to suggest a `setBlockDescription` approach here, but I see that this one has to modify the block to add the editable input field, so in one case it would have a return value and in one case it wouldn't... 🤔 

What happens to `block.description` if the modal functioned editor is turned off? Does it just remain `undefined`, or should we be setting the description in both cases?

Reply:
The description source of truth in sources (i.e. the saved version of the code) is extra state. When we load a project, we load the serialized block from sources. Depending on the modal editor status, we either add a new field and put the description from extra state into it (or it is empty if the description is undefined), or we save the description as `block.description`. Then when we save we take the description from `block.description` or the field and save it as extra state. Both the property and the field only exist while we are in a loaded state of blockly. If we reload and the editor is in a different state (ex. it was off, now it's on or vice versa), that's fine because we will just load the extra state in the appropriate way for the editor state.

So to your question, `block.description` is ephemeral. It only exists while we are in a loaded Blockly state, then it gets turned back into extra state on save.

Let me know if this makes sense!

Reply:
I think if we rename it to `setBlockDescription` I want to keep the logic of setting `block.description` inside that function, since the name implies we are setting something in the method.

Reply:
Sounds good!